src/UglyToad.PdfPig/Util/ColorSpaceDetailsParser.cs

```diff
@@ -98,22 +107,159 @@ internal static class ColorSpaceDetailsParser
+
+                        // WhitePoint is required
+                        if (!DirectObjectFinder.TryGet(second, scanner, out DictionaryToken dictionaryToken) ||
+                            !dictionaryToken.TryGet(NameToken.WhitePoint, out ArrayToken whitePointToken))
+                        {
+                            return UnsupportedColorSpaceDetails.Instance;
+                        }
+
+                        var whitePoint = whitePointToken.Data.OfType<NumericToken>().Select(x => x.Data).ToList();
+
+                        // BlackPoint is optional
+                        IReadOnlyList<decimal> blackPoint = null;
+                        if (dictionaryToken.TryGet(NameToken.BlackPoint, out ArrayToken blackPointToken))
+                        {
+                            blackPoint = blackPointToken.Data.OfType<NumericToken>().Select(x => x.Data).ToList();
+                        }
+
+                        // Gamma is optional
+                        decimal? gamma = null;
+                        if (dictionaryToken.TryGet(NameToken.Gamma, out NumericToken gammaToken))
+                        {
+                            gamma = gammaToken.Data;
+                        }
+
+                        return new CalGrayColorSpaceDetails(whitePoint, blackPoint, gamma);
+                    }
                 case ColorSpace.CalRGB:
-                    return UnsupportedColorSpaceDetails.Instance;
+                    {
+                        if (!TryGetColorSpaceArray(imageDictionary, resourceStore, scanner, out var colorSpaceArray)
+                            || colorSpaceArray.Length != 2)
+                        {
+                            return UnsupportedColorSpaceDetails.Instance;
+                        }
+
+                        var first = colorSpaceArray[0] as NameToken;
+
+                        if (first == null || !ColorSpaceMapper.TryMap(first, resourceStore, out var innerColorSpace)
+                            || innerColorSpace != ColorSpace.CalRGB)
+                        {
+                            return UnsupportedColorSpaceDetails.Instance;
+                        }
+
+                        var second = colorSpaceArray[1];
+
+                        // WhitePoint is required
+                        if (!DirectObjectFinder.TryGet(second, scanner, out DictionaryToken dictionaryToken) ||
+                            !dictionaryToken.TryGet(NameToken.WhitePoint, out ArrayToken whitePointToken))
+                        {
+                            return UnsupportedColorSpaceDetails.Instance;
+                        }
+
+                        var whitePoint = whitePointToken.Data.OfType<NumericToken>().Select(x => x.Data).ToList();
+
+                        // BlackPoint is optional
+                        IReadOnlyList<decimal> blackPoint = null;
+                        if (dictionaryToken.TryGet(NameToken.BlackPoint, out ArrayToken blackPointToken))

```

Comment:
I think for all calls to `DictionaryToken.TryGet` in this class/method we'd need to use the overload/extension which takes `IPdfTokenScanner` as the second argument since it's quite likely these dictionary values will be behind indirect references (e.g. `103 0 R` where object 103 is an array), so for example this call would become:

```suggestion
                        if (dictionaryToken.TryGet(NameToken.BlackPoint, scanner, out ArrayToken blackPointToken))
```

This way it should automatically look up the actual ArrayToken even where it is not directly embedded in the dictionary, which seems to happen frequently. I often forget it myself and then get bug reports where the dictionary contains an indirect reference.